### PR TITLE
Update Changelog for 2.4

### PR DESCRIPTION
**Pull request by daschuer**
This adds another 304 lines to the changelog. 
Thank you for all the work and love that is hidden behind each line! 

This is the raw version of the Changelog with all PRs and almost all Issues referenced. 
I like to merge it to make the next meld down step to a hand-able size visible in a diff. 

Please check also the rendered md version.

### CODE REVIEW

**Comment by mxmilkiib** [comment]
Is missing #12376

**Comment by ronso0** [comment]
> Is missing #12376

No, that was for `main`, not `2.4`

**Comment by mxmilkiib** [comment]
Aaah, so all new features go to main, so it's just in 2.5-alpha onward? I wonder how much there is to go wrong with it, but cool, fair enough if there is a process to be stuck to.

**Comment by daschuer** [comment]
At least the raw version I have started with was auto-generated by the attached script. 
All the sorting and matching with issue numbers is a manual process, which can go wrong. 

In the next step after merge we will remove entries not important to users. That can also go wrong. 
Help is welcome.

**Review by ywwg — APPROVED** [review]
lgtm -- we can create a reduced list from this for the blog post and do language correction and refinement there.

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:443]: both PRs went into 2.3
```suggestion
```

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:449]: The issues is still open, the PR did likely not fix it (relying on @Swiftb0y's [comment](https://github.com/mixxxdj/mixxx/pull/12048#pullrequestreview-1681532979) here)
```suggestion
* MIDI4lights: Give beginTimer callbacks the anonymous function expression form [#12048](https://github.com/mixxxdj/mixxx/pull/12048)
```

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:462]: that one went into 2.3
```suggestion
```

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:475]: was merged to master in 2020
```suggestion
```

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:503]: that one went into 2.3 before 2.3.6 was released

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:627]:
```suggestion
* Fix Icon rendering on HiDPI/retina screens [#12407](https://github.com/mixxxdj/mixxx/pull/12407) [#12361](https://github.com/mixxxdj/mixxx/issues/12361)
```

**Review by ronso0 — CHANGES_REQUESTED** [review]
this is my first run, stopped above the Waveforms section

**Comment by ronso0** [comment]
Found only small issues.
Please remove
`[#4255](https://github.com/mixxxdj/mixxx/pull/4455)`

**Comment by daschuer** [comment]
Thank you for the detailed review!

**Comment by daschuer** [comment]
We had a brief discussion how to proceed with this bulky long changelog. A consensus was to not put much extra hours to it for polishing, because no user will read it line by line. In earlier entries we have removed developer topics. I think I will at least do this to match the style of previous entries after merge. Ideas?

**Comment by ronso0** [comment]
Sute, except mapping dev topics I assume.

File: tools/createChangelog.py
Comment (ywwg) [inline tools/createChangelog.py:1]: can you add a short docstyle comment to describe what this script does?  (does it update the changelog or create it from scratch? when should it be used? etc)

**Review by ywwg — APPROVED** [review]

File: tools/createChangelog.py
Comment (daschuer) [reply tools/createChangelog.py]: Done.

**Comment by daschuer** [comment]
Ready.

**Review by Swiftb0y — APPROVED** [review]
LGTM. waiting for @ronso0 to approve or resolve.

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:745]:
```suggestion
* Allow clearing chains with empty '---' preset [#10859](https://github.com/mixxxdj/mixxx/pull/10859) [#10777](https://github.com/mixxxdj/mixxx/issues/10777)
* Show Effectchain tooltip [#11902](https://github.com/mixxxdj/mixxx/pull/11902) [#10605](https://github.com/mixxxdj/mixxx/issues/10605)
* Center Super knob when loading empty (QuickEffect) chain preset [#12320](https://github.com/mixxxdj/mixxx/pull/12320)
* Read effect data for decks added after initial setup [#12282](https://github.com/mixxxdj/mixxx/pull/12282) [#12277](https://github.com/mixxxdj/mixxx/issues/12277)
* Fix deletion of effect chain presets created in current session [#11886](https://github.com/mixxxdj/mixxx/pull/11886)
* Don't reset "super" and "mix" knob on startup [#11781](https://github.com/mixxxdj/mixxx/pull/11781) [#11773](https://github.com/mixxxdj/mixxx/issues/11773)
* Re-add empty chain preset when resetting to defaults [#11705](https://github.com/mixxxdj/mixxx/pull/11705)
* Add a missing early return [#11809](https://github.com/mixxxdj/mixxx/pull/11809) [#111808](https://github.com/mixxxdj/mixxx/issues/11808)
```

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:824]:
```suggestion
* Windows packaging: Use Azure for signing exe, msi and all dlls with timestamp and sha256
```

File: CHANGELOG.md
Comment (ronso0) [inline CHANGELOG.md:1353]: I remember we came across an issue recently that we wanted to add to 'known issues' (but we didn't think of this place). I added the Changelog label and added it to the 2.4 milestone, but I can't find it anymore...

Maybe someone else remembers what that was.
I think @fwcd and/or @m0dB were involved.

**Review by ronso0 — COMMENTED** [review]
LGTM
Let's remove the redundant 'Effects' prefix.

**Comment by ronso0** [comment]
Thank you, looks good!

**Pull request merged**
